src/node/SyncDish.mjs

```diff
@@ -0,0 +1,196 @@
+/**
+ * @author d98762625 [d98762625@gmail.com]
+ * @copyright Crown Copyright 2018
+ * @license Apache-2.0
+ */
+
+import util from "util";
+import Utils from "../core/Utils";
+import Dish from "../core/Dish";
+import BigNumber from "bignumber.js";
+import log from "loglevel";
+
+/**
+ * Subclass of Dish where `get` and `_translate` are synchronous.
+ * Also define functions to improve coercion behaviour.
+ */
+class SyncDish extends Dish {
+
+    /**
+    * Create a Dish
+    * @param {any} inputOrDish - The dish input
+    * @param {String|Number} - The dish type, as enum or string
+    */
+    constructor(inputOrDish=null, type=null) {
+
+        // Allow `fs` file input:
+        // Any node fs Buffers transformed to array buffer
+        // NOT Buffer.buff, as this makes a buffer of the whole object.
+        if (Buffer.isBuffer(inputOrDish)) {
+            inputOrDish = new Uint8Array(inputOrDish).buffer;
+        }
+
+        super(inputOrDish, type);
+    }
+
+    /**
+     * Apply the inputted operation to the dish.
+     *
+     * @param {WrappedOperation} operation the operation to perform
+     * @param {*} args - any arguments for the operation
+     * @returns {Dish} a new dish with the result of the operation.
+     */
+    apply(operation, args=null) {
+        return operation(this.value, args);
+    }
+
+    /**
+     * Synchronously returns the value of the data in the type format specified.
+     *
+     * @param {number} type - The data type of value, see Dish enums.
+     * @param {boolean} [notUTF8=false] - Do not treat strings as UTF8.
+     * @returns {*} - The value of the output data.
+     */
+    get(type, notUTF8=false) {
+        if (typeof type === "string") {
+            type = Dish.typeEnum(type);
+        }
+        if (this.type !== type) {
+            this._translate(type, notUTF8);
+        }
+        return this.value;
+    }
+
+    /**
+     * alias for get
+     * @param args see get args
+     */
+    to(...args) {
+        return this.get(...args);
+    }
+
+    /**
+     * Avoid coercion to a String primitive.
+     */
+    toString() {
+        return this.get(Dish.typeEnum("string"));
+    }
+
+    /**
+     * What we want to log to the console.
+     */
+    [util.inspect.custom](depth, options) {
+        return this.get(Dish.typeEnum("string"));
+    }
+
+    /**
+     * Backwards compatibility for node v6
+     * Log only the value to the console in node.
+     */
+    inspect() {
+        return this.get(Dish.typeEnum("string"));
+    }
+
+    /**
+     * Avoid coercion to a Number primitive.
+     */
+    valueOf() {
+        return this.get(Dish.typeEnum("number"));
+    }
+
+    /**
+     * Synchronously translates the data to the given type format.
+     *
+     * @param {number} toType - The data type of value, see Dish enums.
+     * @param {boolean} [notUTF8=false] - Do not treat strings as UTF8.
+     */
+    _translate(toType, notUTF8=false) {
```

Comment:
I'm concerned that this will fall out of sync with `core/Dish`. I don't suppose there's any way of overriding `File` with a class that just throws an error when instantiated?

Reply:
Considering the issues that having an asynchronous Dish causes, perhaps it would be worth our while taking a look at this? https://developer.mozilla.org/en-US/docs/Web/API/FileReaderSync

It seems to be supported by most browsers these days. It would mean we could get rid of `SyncDish` entirely.

Reply:
@n1474335  I've spent a few days struggling with this. 

FileReaderSync has to be run inside a worker which have an async interface (onmessage, etc), so that wouldnt really make `_translate` synchronous.

I also experimented with using the async Dish in the node API. In my initial trials, this made it so the wrapper in `src/node/api.mjs` ended up being async (it uses Dish to translate args ready to be consumed by an operation), which meant that the index has to be an async export, which meant that you had to import it with an async dynamic import, which just doesn't feel nice at all.

I'm very much behind what you're aiming for, I just can't figure out how to get there...